### PR TITLE
LPS-143025

### PR DESCRIPTION
**Pull request by linolaoj**
Related Issue: https://issues.liferay.com/browse/LPS-143025

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java [hunk 1]
```diff
@@ -51,7 +51,12 @@ public void register(Registry registry) {
 				DEDataDefinitionFieldLinkUpgradeProcess());
 
 		registry.register(
-			"2.1.1", "2.2.0",
+			"2.1.1", "2.1.2",
+			new com.liferay.data.engine.internal.upgrade.v2_1_2.
+				DEDataDefinitionFieldLinkUpgradeProcess());
+
+		registry.register(
+			"2.1.2", "2.2.0",
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java:59]: @linolaoj, why didn't you create the new upgrade step considering the last available step 2.2.0? Shouldn't we use 2.2.1 instead? Are you doing it due to backport?

Thanks in advance for your clarifications. :slightly_smiling_face:

**Comment by natocesarrego** [comment]
@linolaoj, can you please also check the reported SF failures?

Thanks.

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java (hunk 1 above)
Comment (linolaoj) [reply modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java:59]: Hi @natocesarrego, yes, it is for backporting reasons. I chose 2.1.2 because this needs to run when the user upgrades from 7.3 to 7.4 ga1. But I can add a new step if you believe it is better.

**Comment by linolaoj** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java (hunk 1 above)
Comment (natocesarrego) [reply modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java:59]: Thanks for your clarification, @linolaoj! :slightly_smiling_face: 

@linolaoj, if we keep the existing upgrade steps sequence and create a new upgrade step with number 2.2.1, a micro version, shouldn't it be enough?

According to the Upgrade Team docs, [either minor and micro changes can be backported](https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Schema-version), therefore I believe that we can follow the actual sequence of upgrade steps. If you can't do it for backport limitations then we should keep this pull as it is. :slightly_smiling_face:

**Comment by linolaoj** [comment]
Hi @natocesarrego, I've pushed the changes. Let me know if some other change is needed. 
Thanks!

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/v2_2_1/DEDataDefinitionFieldLinkUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * @author Lino Alves
+ */
+public class DEDataDefinitionFieldLinkUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		alter(
+			DEDataDefinitionFieldLinkTable.class,
+			new AlterColumnType("fieldName", "VARCHAR(255) null"));
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/v2_2_1/DEDataDefinitionFieldLinkUpgradeProcess.java:29]: @linolaoj, sorry for not mention it in the previous review, but, apparently, it's a good upgrade practice check if the column exists before trying to manipulate it, please check [here](https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Choose-the-proper-upgrade-step).

File: modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java [hunk 3]
```diff
@@ -63,6 +63,11 @@ public void register(Registry registry) {
 
 			},
 			new UpgradeCTModel("DEDataDefinitionFieldLink", "DEDataListView"));
+
+		registry.register(
+			"2.2.0", "2.2.1",
+			new com.liferay.data.engine.internal.upgrade.v2_2_1.
+				DEDataDefinitionFieldLinkUpgradeProcess());
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/upgrade/DEServiceUpgrade.java:70]: @linolaoj, I was checking the 7.3.x code and I realized that the Upgrade Step 2.2.0 isn't available there, in light of this perhaps you'll need to use your previous solution, what do you think? Another alternative would be also backport the Upgrade Step 2.2.0, but it seems to be out of scope of the issue you're fixing.

**Comment by linolaoj** [comment]
Hi @natocesarrego , I've changed back to the previous version of the fix. Let me know if any other change is needed.
Happy Christmas!

**Comment by natocesarrego** [comment]
For you too, @linolaoj! :smile: :christmas_tree: :santa: 

Have you had the chance to see this [last comment](https://github.com/liferay-forms/liferay-portal/pull/1264#discussion_r773115699)? Apparently, we should use the `hasColumn` validation.

Thanks!

**Comment by linolaoj** [comment]
Hi @natocesarrego, sorry I forgot to include that change. I've pushed the column check now. Let me know if any other change is needed.

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:reevaluate:1257401_1173

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by carolmariaabb** [comment]
ci:reevaluate:1297401_1196

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by natocesarrego** [comment]
ci:forward

**Comment by liferay-forms** [comment]
Hey @linolaoj, can you please check if these failures are related to your changes? Thanks

**Comment by natocesarrego** [comment]
@linolaoj, do you mind to rebase your pull and resend?

Unfortunately, this pull seems to be stuck due to CI. :disappointed: 

Thanks in advance

**Comment by linolaoj** [comment]
rebased at https://github.com/liferay-forms/liferay-portal/pull/1350
